packages/pyright-internal/src/analyzer/codeFlowEngine.ts

```diff
@@ -1583,7 +1583,15 @@ export function getCodeFlowEngine(
             return true;
         }
 
-        return evaluator.isAfterNodeReachable(functionType.details.declaration.node);
+        // This call is very expensive because it requires a full code flow analysis
+        // of this function and all (unannotated) functions that it calls, recursively.
+        // Rather than incurring this expense, we assume here that all unannotated
+        // functions are not NoReturn functions. To reverse this decision, uncomment
+        // the following line.
+
+        // return evaluator.isAfterNodeReachable(functionType.details.declaration.node);
+
```

Comment:
should we keep it for user code?

Reply:
or maybe a setting

Reply:
No, type evaluation should be consistent, not depend on whether it's user code or not and should not depend on a setting.